modules/cert-manager-acme-dns01-aws.adoc

```diff
@@ -0,0 +1,113 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-operator-issuer-acme.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-acme-dns01-aws_{context}"]
+= Issuing certificates by using the DNS01 challenge
+
+// TODO work on title
+
+You can use {cert-manager-operator} to set up an ACME issuer to solve DNS01 challenges. This procedure shows how to solve DNS01 challenges with Amazon Route 53.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+* You have set up an IAM role for Amazon Route 53. For more information, see link:https://cert-manager.io/docs/configuration/acme/dns01/route53/[Route53] in the upstream cert-manager documentation.
++
+[NOTE]
+====
+For the {cert-manager-operator}, you must provide an `accessKeyID` and `secretAccessKey` to specify the credentials.
+====
+
+.Procedure
+
+. Configure the nameserver settings for the DNS01 self check:
+
+.. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+.. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=1.1.1.1:53' <2>
+      - '--dns01-recursive-nameservers-only'       <3>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS01 self check.
+<3> Specify to only use recursive nameservers.
++
+// TODO: what exactly does this mean? That it uses caching instead of checking the authoritative nameservers every time? Might cause it to take longer it says why? (every time)?
```

Comment:
@thejasn @swghosh Can you clarify what this "--dns01-recursive-nameservers-only" really means? The upstream docs weren't clear to me.

Reply:
It basically forces cert-manager to not search for authoritative nameservers (associated with that domain) at all and only use the recursive nameservers that  are provided. The part where it say this might take longer due to caches, even I'm not very clear on. Will try to get back on this.

Reply:
Yeah it seemed like if it was caching it, then it should actually take less time. I can leave this information out for now, but this would be good to state if we know one way or the other.

Reply:
Not worrying about this for now. If you find that we should say anything about time it takes with vs without this option, let me know and we can add later.